apm-agent-core/src/test/java/co/elastic/apm/agent/MockTracer.java

```diff
@@ -81,6 +84,47 @@ public static ElasticApmTracer createRealTracer(Reporter reporter, Configuration
             .build();
     }
 
+    /**
+     * If an instrumentation has already been initialized by some other test, returns the static
+     * {@link co.elastic.apm.agent.bci.ElasticApmInstrumentation#tracer}.
+     * Otherwise, Creates a real tracer with a {@link MockReporter} and a mock configuration which returns default
+     * values that can be customized by mocking the configuration.
+     */
+    public static synchronized MockInstrumentationSetup getOrCreateInstrumentationTracer() {
+
+        ElasticApmTracer tracer = ElasticApmInstrumentation.tracer;
+        if (tracer == null || tracer.getState() == ElasticApmTracer.TracerState.STOPPED) {
```

Comment:
why does a stopped tracer should trigger creating a new one ? Is that triggered for example if a test leaves the tracer in a stopped state ? Couldn't we just resume it when it's the case ?

What is a bit confusing here is that we have a call to `TracerInternalApiUtils.resumeTracer(tracer);` in the `else` block.

Reply:
This is according to the way we defined the `STOPPED` state [as irreversible](https://github.com/elastic/apm-agent-java/blob/96b7d60e4edcb39242ab5d7922111d52fd9c1023/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java#L784).

> Is that triggered for example if a test leaves the tracer in a stopped state ?

Yes, for example - [this one](https://github.com/elastic/apm-agent-java/blob/842185664456651f1d6a99602d21a7c052ddf111/apm-opentracing/src/test/java/co/elastic/apm/opentracing/OpenTracingBridgeCloseTest.java#L48).